lib/node_runner_checker.go

```diff
@@ -275,9 +275,21 @@ func CheckNodeRunnerHandleBallotVotingHole(c sebakcommon.Checker, args ...interf
 			return
 		}
 		// check, source is same with the `BlockTransaction` of tx's checkpoint
-		if tx.B.Source != bt.Source {
+		var checkPoint [2]string
+		if checkPoint, err = sebakcommon.ParseCheckpoint(tx.B.Checkpoint); err != nil {
 			return
 		}
+
+		// The transaction was invoked by the account
+		if checkPoint[0] == checkPoint[1] {
+			if tx.B.Source != bt.Source {
+				return
+			}
+		} else { //The transaction was invoked by another account
+			if _, found := sebakcommon.InStringArray(bt.Targets, tx.B.Source); !found {
```

Comment:
Just curious, why is checking the target address needed?
If the purpose of this pr is to fix bug, then only the code of this part is enough.
```
		if checkPoint[0] == checkPoint[1] {
			if tx.B.Source != bt.Source {
				return
			}
		}
```
I already checked #150 scenario(thanks @kfangw) with commented this part like,
```
} else { //The transaction was invoked by another account
			// if _, found := sebakcommon.InStringArray(bt.Targets, tx.B.Source); !found {
			// 	return
			// }
		}
```

I figure, remaining target information is using more resources so we should be careful :)

Reply:
I second this. I thought we were not supposed to check the target address ?